### ISSUE TITLE
error: protect(): protection stack overflow 

### ISSUE THREAD

**Issue by HemanthK** [dirkschumacher/ompr#181]
I am using the Package in R on 4 Million rows and getting the following error.
error: protect(): protection stack overflow
The Data is being stored in a CSV file and it runs for 30 Minutes adds the constraints before throwing the error.
If I reduce the data it works fine
The Code is as below any suggestions will help.
[omprcode.txt](https://github.com/dirkschumacher/ompr/files/1864496/omprcode.txt)

**Comment by dirkschumacher** [comment]
More info here: http://reprex.tidyverse.org/articles/reprex-dos-and-donts.html

**Comment by dirkschumacher** [comment]
Also, if you don't mind, you could try the `MILPModel` backend. It is really fast, but has slightly different semantics. A bit experimental, but will be included in the next CRAN release.

**Comment by dirkschumacher** [comment]
I drafted some articles: https://github.com/dirkschumacher/ompr/tree/master/vignettes

Take a look at those atricles ending with `milp`. Happy to hear feedback :)

**Comment by HemanthK** [comment]
Trying to run the samples mentioned in your articles and getting the following error 
Error in MILPModel() : could not find function "MILPModel"

**Comment by rickyars** [comment]
@dirkschumacher (somewhat) unrelated: What's the ETA on 0.8? I really want to try `MILPModel` but I'm not allowed to install packages from GitHub (it's also blocked by our firewall).

**Comment by dirkschumacher** [comment]
@HemanthK you have to use the github version of the package.
@rickyars I'll try to push it in the coming days (can't promise anything though)

**Comment by dirkschumacher** [comment]
I upaded the docs for the development version: https://dirkschumacher.github.io/ompr/dev/articles/index.html

**Comment by dirkdegel** [comment]
Hi everyone,

I also get the error message when using the `MIPModel()` function for a problem with a bit more then 18000 variables and 1050 constraints:
`Error: protect(): protection stack overflow`

Basically, the `MIPModel` function is not able to generate the matrix and run into this error. 

I was wondering if there is meanwhile a solution. I found this, which is not related to the OMPR package but to the error message:
[https://stackoverflow.com/questions/32826906/how-to-solve-protection-stack-overflow-issue-in-r-studio](https://stackoverflow.com/questions/32826906/how-to-solve-protection-stack-overflow-issue-in-r-studio)

I tried to set `options(expressions = 5e5)` and it worked (I checked with `getOption("expressions")` ) but it still gives me the same error.

For me it is also not clear if the problem is related to the OMPR package (and to the `MIPModel` function in particular, e.g., could be solved by using `MILPModel` function) or to R itself.

Thanks a lot everybody!
